lib/deep-map.js

```diff
@@ -8,20 +8,32 @@ function filterError (input) {
   }
 }
 
-const deepMap = (input, handler = v => v, path = ['$'], seen = new Set([input])) => {
+const deepMap = (
+  input,
+  handler = v => v,
+  path = ['$'],
+  seen = new Set([input]),
```

Comment:
🤔 If `input` is an array, should we avoid doubly nesting it?

Reply:
Related, would it be easier to reason about this logic if we had the exported function do something like:

```
module.exports = function (input, etc) {
  if (Array.isArray(input)) {
    return input.map(e => deepMap(e, ...etc))
  }
  return deepMap(input, ...etc)
}
```

Reply:
`Set` takes an array and an item in that array is an array, and we do a match on the instance of that array (input), to check if it is `seen`, I think `seen` is correct the way it is.